pkg/trace/agent/agent.go

```diff
@@ -460,3 +460,8 @@ func newEventProcessor(conf *config.AgentConfig) *event.Processor {
 
 	return event.NewProcessor(extractors, conf.MaxEPS)
 }
+
+// SetGlobalTags sets global tags to the agent configuration
```

Comment:
Is there any risk of a race here? Seem on the edge...

Reply:
It's indeed possible to receive logs before calling this method. Maybe I can add a comment to make it clearer ?

Reply:
Sorry but I'm not sure what you mean. The question is whether there is a risk of a race condition, where the config object gets initialised and this method gets called at the same time, resulting in a panic and a crash.

Reply:
Yes I see. In the way we are using it, we are 100% sure that config has already been initialized

Reply:
Still, this is bad API and exposes that risk. I don't feel comfortable with this solution. You could instead just set the tags directly on the config object in `cmd/serverless/main.go` around this line: https://github.com/DataDog/datadog-agent/blob/eabed0840342392f9cd118ecdb9f58ea930a8f8b/cmd/serverless/main.go#L355
 
What am I missing?

Reply:
Yes we did that before and we needed to know tags in advance, during the log agent setup, which is before the actual first lambda invocation. To retrieve those tags, we needed a STS call. We have removed that for performance issues. Instead we are waiting for the first invocation to occur, which give us the ARN. From this ARN we can get tags without having to call STS. Then we apply those tags to the agent config + we make sure that no traces go through the agent before setting those tags

Reply:
Hi @gbbr did you have the chance to look at my previous comment? Let me know what do you think about it.